src/Core.js

```diff
@@ -306,6 +306,10 @@ function getLocation(url){
     }
     // #endif
     var m = url.toLowerCase().match(reURI);
+    //if the origin is non standard url, such as chrome extensions
+    if(!m) {
```

Comment:
This will address the issue as I see it as well.  I defer to @oyvindkinsey on whether a return of `''` is appropriate, or if he'd prefer this to be an Error situation like the two cases above it.  (I would have probably done what you did here.)